### DISCUSSION TITLE
Set hWnd as owner for avalonia window

### DISCUSSION THREAD

**Discussion by patrickiel** [AvaloniaUI/Avalonia#9015] [Q&A] [answered]
Hello,
I want to use Avalonia for an add in. The host application is accessible via API and I can get its WindowHandle (hWnd).
How do I make a modal dialog when I have only the Window Handle (hWnd)? Window.ShowDialog(..) accepts only the type "Avalonia.Controls.Window" as owner.

Thanks!

**Comment by kekekeks** [comment, marked as answer]
Just use `Show()` and then change GWL_HWNDPARENT via P/Invoke

**Reply by patrickiel** [reply]
Thank you!
